src/modules/analysis/analysisEditContainer.js

```diff
@@ -0,0 +1,77 @@
+import inject from 'react-rxjs/dist/RxInject';
+import React from 'react';
+import analysisAddForm from './analysisAddForm';
+import AnalysisEditComponent from './AnalysisEditComponent';
+import store$, { loadAnalysis$, getAnalysisTypesForCollection$ } from './analysisStore';
+import Analysis from '../../models/analysis';
+import { makeUrlAware } from '../app/appSession';
+import flowRight from 'lodash/flowRight';
+import mount from '../../shared/mount';
+import { toPromise } from '../../shared/util';
+
+const { form$, updateForm$, loadForm$ } = analysisAddForm;
+
+const data = {
+  appSession$: { type: React.PropTypes.object.isRequired },
+  store$,
+  form$
+};
+
+const commands = {
+  updateForm$,
+  loadForm$,
+  getAnalysisTypesForCollection$,
+  loadAnalysis$
+};
+
+const props = {
+  editAnalysisEvent: toPromise(Analysis.editAnalysisEvent()),
+  loadAnalysisForForm: toPromise(Analysis.getAnalysisWithDeatils())
+};
+
+const callLoadAnalysisForForm = (loadAnalysisForForm, appSession, params, loadForm) =>
+  loadAnalysisForForm({
+    museumId: appSession.getMuseumId(),
+    id: params.analysisId,
+    collectionId: appSession.getCollectionId(),
+    token: appSession.getAccessToken()
+  }).then(analysis => {
+    console.log(analysis);
+    const dataForForm = Object.keys(analysis).reduce(
+      (obj, attributeName) => [
+        ...obj,
+        { name: attributeName, defaultValue: analysis[attributeName] }
+      ],
+      []
+    );
+    console.log(dataForForm);
+    loadForm(dataForForm);
+  });
+
+export const onMount = (
```

Comment:
you can very easily test this method in analysisEditContainer-test.js. Just pass in spies. And check if spies is calledOn after calling onMount. Remember to pass in callLoadAnalysisForForm also, dont call it statically